tests/test_models.py

```diff
@@ -26,3 +49,92 @@ def test_without_conditions(model):
     gen.fit(X, C=None)
     X_gen = gen.sample(C=n)
     assert X_gen.shape == X.shape
+
+
+logdets = ['exact', 'fixed', 'unbias']
+res_wrappers = [ResidualUnconditional, ResidualConditional]
+@pytest.mark.parametrize("wrapper_model,logdet", list(itertools.product(res_wrappers, logdets)))
+def test_resflow(wrapper_model, logdet):
+    is_cond = (wrapper_model == ResidualConditional)
+
+    n = 100
+    X = torch.from_numpy(np.random.normal(size=(n, 5))).to(torch.float32)
+    y = torch.from_numpy(np.random.normal(size=(n, 3))).to(torch.float32)
+    len_y = y.shape[1]; len_X = X.shape[1]
+
+    flow_args_dict = {
+        'var_dim': len_y if is_cond else len_y + len_X,
+        'cond_dim': len_X if is_cond else None,
+        'hid_dim': 16,
+        'n_block_layers': 3,
+        'n_layers': 3,
+        'spnorm_coeff': 0.95,
+        'n_backward_iters': 100,
+        'logdet': logdet,
+    }
+
+    flow = ResidualFlowModel(**flow_args_dict)
+
+    start_lr = 1e-1
+    final_lr = 1e-2
+    n_epochs = 100
+    sched_lambda = (final_lr / start_lr) ** (1 / n_epochs)
+
+    optim = torch.optim.Adam(flow.parameters(), lr=start_lr, weight_decay=1e-5)
+    sched = torch.optim.lr_scheduler.ExponentialLR(optim, sched_lambda)
+    wrapper = wrapper_model(flow, optim, n_epochs=n_epochs, batch_size=100, scheduler=sched)
```

Comment:
Do not use any wrappers, please. Please, look at the model interfaces in the main branch of Probaforms.